sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/SwitchCasesMinimumThree.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.Rules.Common;
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class SwitchCasesMinimumThree : SwitchCasesMinimumThreeBase
+    {
+        private const string MessageFormat = "Replace this 'Select' statement with 'If' statements to increase readability.";
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var selectNode = (SelectBlockSyntax)c.Node;
+                    if (!HasAtLeastThreeLabels(selectNode))
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, selectNode.SelectStatement.SelectKeyword.GetLocation()));
+                    }
+                },
+                SyntaxKind.SelectBlock);
+        }
+
+        private static bool HasAtLeastThreeLabels(SelectBlockSyntax node)
+        {
+            return node.CaseBlocks.Sum(caseBlock => caseBlock.CaseStatement.Cases.Count) >= 3;
```

Comment:
Could you check the C# implementation to see if we do count multiple values on the same case? If not, could you update (and create a ticket) so that we have the same behavior for C# and VB.Net

Reply:
`Case 1, 2` is the equivalent of `case 1: case 2:` in C#, which is already counted as 2 cases. Was that what you has in mind when saying *"if we do count multiple values on the same case"*?